contracts/liquidity_hub/whale_lair/src/queries.rs

```diff
@@ -167,38 +188,24 @@ pub(crate) fn query_weight(
         total_bond_weight = total_bond_weight.checked_add(bond.weight)?;
     }
 
-    // for each of the unique denoms, access the unbondings
-    // and reduce the weight of the unbonding to the total_bond_weight
-    // TODO: REVIEW THIS WITH J
-    for denom in unique_denoms {
-        let unbonding = query_unbonding(deps, address.to_string(), denom.to_string(), None, None)?;
-        for bond in unbonding.unbonding_requests {
-            // TODO: Review this with J
-            if bond.timestamp.plus_nanos(config.unbonding_period.u64()) < timestamp {
-                let weight = get_weight(
-                    timestamp,
-                    bond.weight,
-                    bond.asset.amount,
-                    config.growth_rate,
-                    bond.timestamp,
-                )?;
-                total_bond_weight = total_bond_weight.checked_sub(weight)?;
-            }
-        }
-    }
+    let mut global_index = if let Some(global_index) = global_index {
+        global_index
```

Comment:
This was the magic that fixed the rewards, as the global index contains a snapshot of the global weight and the bonded assets when the epoch was created <3